python/pyiceberg/utils/datetime.py

```diff
@@ -67,6 +67,11 @@ def time_to_micros(time_str: str) -> int:
     return (((t.hour * 60 + t.minute) * 60) + t.second) * 1_000_000 + t.microsecond
 
 
+def time_object_to_micros(t: time) -> int:
+    """Converts an datetime.time object to microseconds from midnight."""
+    return int(t.hour * 60 * 60 * 1e6 + t.minute * 60 * 1e6 + t.second * 1e6 + t.microsecond)
```

Comment:
I think this should use the same logic as line 67 above. There are a couple of good things about that form:
1. It stays in `int` and doesn't introduce any floating point values. That avoids the cast and, more importantly, avoids any floating point math that may introduce errors.
2. It's easier to read and understand that it is correct.